tests/qm-sanity-test/test.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash -euvx
+
+# shellcheck disable=SC1091
+source ../e2e/lib/utils
+
+# Verify qm is up and running
+check_qm_is_up(){
+    if [ "$(systemctl is-active qm)" == "active" ]; then
+        info_message "check_qm_is_up(): qm is active"
+    else
+        info_message "check_qm_is_up(): qm is not active"
+        exit 1
+    fi
+}
+
+# Verify bluechi nodes are connected
+check_bluechi_nodes_connected(){
+    expected_result="Connected to controller as 'qm.localrootfs'"
+    if [[ "$(podman exec -it qm bash -c "systemctl status bluechi-agent" \
+    | tail -2)" =~ ${expected_result} ]]; then
+        info_message "check_bluechi_nodes_connected: \
+        QM bluechi-agent is connected to controller"
+    else
+        info_message "check_bluechi_nodes_connected: \
+        QM bluechi-agent is not connected to controller"
+        exit 1
+    fi
+
+    expected_result="Connected to controller as 'localrootfs'"
+    if [[ "$(systemctl status bluechi-agent | tail -2)" =~ ${expected_result} ]]; then
+        info_message "check_bluechi_nodes_connected: \
+        ASIL bluechi-agent is connected to controller"
+    else
+        info_message "check_bluechi_nodes_connected: \
+        ASIL bluechi-agent is not connected to controller"
+        exit 1
+    fi
+}
+
+# Verify podman run and exec container inside qm with service file
+check_podman_run_and_exec(){
+    info_message "check_podman_run_and_exec(): \
+    prepare quadlet files for qm-sanity-test.container"
+    cat > "/etc/qm/containers/systemd/qm-sanity-test.container" <<EOF
+[Unit]
+Description=The sleep container
+After=local-fs.target
+
+[Container]
+Image=registry.access.redhat.com/ubi9-minimal:latest
+Exec=sleep 1000
+
+[Install]
+# Start by default on boot
+WantedBy=multi-user.target default.target
+EOF
+
+    info_message "check_podman_run_and_exec(): qm-sanity-test reload & restart"
+    exec_cmd "podman exec qm systemctl daemon-reload"
+    exec_cmd "podman exec qm systemctl start qm-sanity-test"
+    exec_cmd "podman exec qm systemctl status qm-sanity-test | grep -i started"
+
+    exec_cmd "podman exec qm podman run alpine echo hello"
+}
+
+# Verify /var partition exist
+check_var_partition_exist(){
+    if [[ "$(podman exec -it qm bash -c "ls /var")" =~ "No such file or directory" ]]; then
+        info_message "check_var_partition_exist: /var does not exist inside QM"
+        exit 1
+    else
+        info_message "check_var_partition_exist: /var exists inside QM"
+    fi
+}
+
+
+check_qm_is_up
```

Comment:
Consider checking the following
check_qm_podman_ok
`podman exec qm bash -c "podman info"` 
Completed exit 0

check_bluechi_controller_ok
`systemctl is-active bluechi-controller`

I would also add check that /var is on separate partition

Reply:
done